### ISSUE TITLE
Using Asyc with minimunInput > 0, all selected items get removed from the text area when you clear the search inputValue.

### ISSUE THREAD

**Issue by yanmaoyuan** [JedWatson/react-select#1220]
### Thanks for using react-select!

If you are reporting an error please include a test case that demonstrates the issue you're reporting!
This is very helpful to maintainers in order to help us see the issue you're seeing.

Here is a Plunker you can fork that has react-select loaded and supports JSX syntax:
https://plnkr.co/edit/HTmtER9AMNcPoWhXV707?p=preview

You may also find the [online Babel tool](https://babeljs.io/repl/) quite helpful if you wish to use ES6/ES7 syntax not yet supported by the browser you are using.

**Comment by yanmaoyuan** [comment]
Steps to produce:
1. Use Asyc fetching options
2. set minimum input to 2 (anything greater than 0 will do)
3. enter some input value into the search box, and select a few items from the dropdown
4. backspace to remove your search input value

expected: the selected items remains
actual: the selected items are removed as soon as inputValue.length < minimumInput

**Comment by bvaughn** [comment]
FYI I am working on a refactor for `Async` that will invalidate this issue. (See PR #1226)

I plan to deprecate the `minimunInput` property. I believe it causes more complication than it's worth and is better handled by application logic. For example, you can enforce a similar behavior in a more customizable way in your application code like so:

```js
const MINIMUM_INPUT_LENGTH = 2

let previousLoadPromise

function loadOptions (inputValue) {
  if (inputValue && inputValue.length > MINIMUM_INPUT_LENGTH) {
    previousLoadPromise = fetch(...) // load options based on inputValue
  if (!previouslyLoadedOptions) {
    previousLoadPromise = fetch(...) // load initial / default options
  }

  // else return the previously-loaded options
  // this way, input text that's shorter than 2 chars won't reset options

  return previousLoadPromise
}
```

**Comment by bvaughn** [comment]
Resolved via PR #1226.

Look for an RC2 to come out (hopefully) sometime today with this change.
